### PR TITLE
Improved Error Handling

### PR DESCRIPTION
**Pull request by TonioF**
This PR is in reaction to repeated observations that xcube gen2 would return rather cryptic 500 errors. On cumbersome investigation, it often turned out that the error was not on server-side, but that the problem was a bad request which resulted in errors that were not handled (correctly).

Checklist:

* ~[ ] Add unit tests and/or doctests in docstrings~
* ~[ ] Add docstrings and API docs for any new/modified user-facing classes and functions~
* ~[ ] New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

Remember to close associated issues after merge!

### CODE REVIEW

File: xcube/core/gen2/local/opener.py [hunk 1]
```diff
@@ -59,41 +62,47 @@ def open_cube(self, input_config: InputConfig) -> TransformedCube:
...
+                dataset = opener.open_data(input_config.data_id,
+                                           **open_params,
+                                           **dataset_open_params)
+                observer.worked(1)
+
+            except DataStoreError as dse:
+                tb = traceback.format_tb(dse.__traceback__)
+                raise CubeGeneratorError(f'{dse}',
+                                         status_code=400,
+                                         remote_traceback=tb)
```
Comment (forman) [inline xcube/core/gen2/local/opener.py:105]: `remote_traceback` is not appropriate in package `local`

File: xcube/core/gen2/local/opener.py [hunk 2]
```diff
@@ -59,41 +62,47 @@ def open_cube(self, input_config: InputConfig) -> TransformedCube:
...
+
+                open_params_schema = opener.get_open_data_params_schema(
+                    input_config.data_id
+                )
 
-            observer.worked(1)
+                dataset_open_params = {
+                    k: v for k, v in cube_params.items()
+                    if k in open_params_schema.properties
+                }
```
Comment (forman) [inline xcube/core/gen2/local/opener.py:92]: I do not understand any of the changes above. What's the reason for this?

Comment (TonioF) [reply xcube/core/gen2/local/opener.py:92]: This is all about catching DataStoreErrors. As there are several instances where one could be raised, I wrapped several lines of a code in a try block. You could think about moving these lines to a dedicated method, however, I did not want to do this as it is also wrapped in a with-block.

File: xcube/core/gen2/local/describer.py [hunk 3]
```diff
@@ -71,10 +71,8 @@ def _describe_dataset(self, input_config: InputConfig) \
         try:
             descriptor = opener.describe_data(input_config.data_id,
                                               data_type=DATASET_TYPE)
-        except DataStoreError:
-            raise CubeGeneratorError(f'Data store '
-                                     f'"{input_config.store_id}" '
-                                     f'does not support datasets',
+        except DataStoreError as dse:
+            raise CubeGeneratorError(f'{dse}',
                                      status_code=400)
```
Comment (forman) [inline xcube/core/gen2/local/describer.py:76]: Keep original traceback:

```suggestion
            raise CubeGeneratorError(f'{dse}',
                                     status_code=400) from dse
```

File: CHANGES.md [hunk 4]
```diff
@@ -8,6 +8,8 @@
   value of `base_dataset_id`, instead of being copied as-is.
   This can save large amounts of storage space. (#617)
 
+* `xcube gen2` returns more expressive error messages
```
Comment (forman) [inline CHANGES.md:11]:
```suggestion
* `xcube gen2` returns more expressive error messages.
```

**Review by forman — CHANGES_REQUESTED** [review]

**Review by AliceBalfanz — APPROVED** [review]

**Pull request merged**